exonum/src/storage/hash.rs

```diff
@@ -31,3 +43,48 @@ impl<T: CryptoHash> UniqueHash for T {
         CryptoHash::hash(self)
     }
 }
+
+/// Convenient method to obtain prefixed value of `StorageValue`.
+pub fn hash_leaf<V: StorageValue>(value: V) -> Hash {
+    let value_bytes = value.into_bytes();
+    hash_with_prefix(LEAF_TAG, &value_bytes)
+}
+
+/// Convenient method to obtain prefixed value of `Hash`.
+pub fn hash_one(h: &Hash) -> Hash {
+    hash_with_prefix(NODE_TAG, h.as_ref())
+}
+
+/// Convenient method to obtain prefixed value of concatenation of two hashes.
+pub fn hash_pair(h1: &Hash, h2: &Hash) -> Hash {
+    let mut hash_bytes = [0u8; HASH_SIZE * 2 + PREFIX_SIZE];
```

Comment:
Do we have a final decision on whether to use `HashStream`? (I'm fine with either of choices.)

Going with the previous comment about an enum with tags, it seems convenient to have the interface for its hashing:

```rust
#[repr(u8)]
pub enum Tag {
    Node = 2,
    // ...
}

impl Tag {
    // if we use streams
    pub fn hash_stream(&self) -> HashStream;
    // or if we don't
    pub fn hash(&self, values: impl Iterator<Item = &[u8]>) -> Hash;
}
```

To go even further, this API could be private, and users may use more high-level abstractions:

```rust
impl Tag {
    fn hash_stream(&self) -> HashStream;

    pub fn hash_node(left_hash: &Hash, right_hash: &Hash) -> Hash {
        Tag::Node
            .hash_stream()
            .update(left_hash.as_ref())
            .update(right_hash.as_ref())
            .hash()
    }
}
```

Reply:
We decided to use `HashStream`.